### PR TITLE
Stop using DI for the session factory

### PR DESCRIPTION
**Pull request by andreasohlund**
cc @timbussmann @danielmarbach

### CODE REVIEW

File: samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs [hunk 1]
```diff
@@ -18,16 +18,11 @@ static void Main()
...
-        endpointConfiguration.RegisterComponents(
-            registration: components =>
-            {
-                components.ConfigureComponent(
-                    componentFactory: builder => new MySessionProvider(),
-                    dependencyLifecycle: DependencyLifecycle.SingleInstance);
-            });
         var pipeline = endpointConfiguration.Pipeline;
-        pipeline.Register(typeof(MyUowBehavior), "Manages the session");
+        pipeline.Register(_ => new MyUowBehavior(sessionProvider), "Manages the session");
```
Comment (danielmarbach) [inline samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs]: If I recall correctly you don't need the lambda overload there is one where you can pass in the behavior instance directly

Comment (andreasohlund) [inline samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs]: Ah, didn't know. Fixed!

File: samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs [hunk 2]
```diff
@@ -18,16 +18,11 @@ static void Main()
         endpointConfiguration.EnableInstallers();
         endpointConfiguration.SendFailedMessagesTo("error");
 
+        var sessionProvider = new MySessionProvider();
```
Comment (SimonCropp) [inline samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs]: should this be inside the region?

Comment (andreasohlund) [inline samples/pipeline/unit-of-work/Core_6/Endpoint/Program.cs]: 👍 done!

**Comment by danielmarbach** [comment]
LGTM

**Comment by timbussmann** [comment]
nice :)

**Pull request merged**